src/connectors/autocomplete/__tests__/connectAutocomplete-test.ts

```diff
@@ -485,6 +485,62 @@ search.addWidgets([
     });
   });
 
+  describe('getWidgetRenderState', () => {
+    test('returns the widget render state', () => {
+      const renderFn = jest.fn();
+      const unmountFn = jest.fn();
+      const createAutocomplete = connectAutocomplete(renderFn, unmountFn);
+      const autocomplete = createAutocomplete({});
+
+      const renderState1 = autocomplete.getWidgetRenderState(
+        createInitOptions()
+      );
+
+      expect(renderState1).toEqual({
+        currentRefinement: '',
+        indices: [],
+        refine: undefined,
+        widgetParams: {},
+      });
+
+      autocomplete.init!(createInitOptions());
+
+      const renderState2 = autocomplete.getWidgetRenderState(
+        createRenderOptions()
+      );
+
+      expect(renderState2).toEqual({
+        currentRefinement: '',
+        indices: expect.any(Array),
+        refine: expect.any(Function),
+        widgetParams: {},
+      });
+    });
+
+    test('returns the widget render state with a query', () => {
+      const renderFn = jest.fn();
+      const unmountFn = jest.fn();
+      const createAutocomplete = connectAutocomplete(renderFn, unmountFn);
+      const autocomplete = createAutocomplete({});
+      const helper = algoliasearchHelper(createSearchClient(), 'indexName', {
+        query: 'query',
+      });
+
+      autocomplete.init!(createInitOptions());
+
+      const renderState = autocomplete.getWidgetRenderState(
+        createRenderOptions({ helper })
+      );
+
+      expect(renderState).toEqual({
+        currentRefinement: 'query',
+        indices: expect.any(Array),
```

Comment:
I think we should assert what's actually in this indices array. In your case it's probably an empty array?

Reply:
https://github.com/algolia/instantsearch.js/pull/4561/commits/924eabc7c18a81d9c6e156c5cd50c96ac5877617 what do you think?